Arvato RSS release 2.0.0

sergeysikachev opened the pull request:
- Developer(s): @sergeysikachev

- Peer Reviewer:

- Ticket: URL_HERE

- Project PR: URL_HERE

- Academy PR: URL_HERE (for feature documentation and if needed migration guides and project integration guide)

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department
- [ ] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraints         |
   :--------------------- | :------------------------ | :--------------------- |
   ArvatoRss              |       major                 |                       |

#### Release Notes

- Arvato Rss  Risk Check functionality and tests refactoring after additional code review.
- Added DeliveryAddress to data send within RiskCheck request.
- Added support for address validation in RiskCheck response.
-----------------------------------------

#### Bundle ArvatoRss

_**Major:** Backwards-compatible feature_

Def of done (by responsible developer):
- [ ] All changes are backward-compatible. Outdated code is marked as deprecated
- [ ] The change is isolated and not mixed with any cleanups or other changes
- [ ] New code fits to the architecture rules
- [ ] All new or changed business logic is covered by functional tests (in Zed business layer)

##### Change log

Initial Release:

- Arvato Rss  Risk Check functionality and tests refactoring after additional code review.
- Added DeliveryAddress to data send within RiskCheck request.
- Added support for address validation in RiskCheck response.

Commit:
Add readme file.

Commit:
Update scrutinizer config.

Commit:
Remove repositories from composer.json.

Commit:
Move API urls from config.dist.php to documentation.

Commit:
Rename iso3166 converter.

Commit:
Remove comment from constants class.

Commit:
Rename ARVATORSS_AUTHORISATION config key

Commit:
Remove password comment from transfer xml.

Commit:
Catch exception earlier if soapfault is received.

Commit:
Remove redundant comments and change params types to interfaces

Commit:
Use is_soap_fault instead of instanceof.

Commit:
Segeregate interface with constants.

Commit:
Remove empty line.

Commit:
Move header creation to a separate method.

Commit:
Remove communication factory.

Commit:
Remove redundant helpers.

Commit:
Pass quote from data provider to test method instead of ArrayObject.

Commit:
Handle Service Unavailable exception.

Commit:
Add header licenses to autogenerated classes.

Commit:
Remove autoload sections from composer.json.

Commit:
Restore autoload section in composer.json.

Commit:
Add missing header licenses.

Commit:
Remove unit suite config.

Commit:
Add unit suite config.

Commit:
Fix test running and add ArvatoRss namespace to codeception config.

Commit:
Tests fixes.

Commit:
Fix tests.

dereuromark commented on README.md:
```diff
@@ -0,0 +1,18 @@
+# ArvatoRSS Bundle
```
module

also: ArvatoRss

sergeysikachev replied:
unconsciously wanted to emphasise :)

dereuromark commented on README.md:
````diff
@@ -0,0 +1,18 @@
+```
+
+## Documentation
+[ArvatoRSS documentation](https://academy.spryker.com/developing_with_spryker/3rd-party_integration/arvatorss/integration_arvato_rss.html)
````
Same here.

Commit:
Update readme.

Commit:
Update use statement.

Commit:
Fix return value

Commit:
Add spryker repository in composer config via scrutiniser config.

Commit:
Fix scrutiniser config.

Commit:
Update scrutiniser config.

profuel committed:
ECO-807 Travis CI setup

profuel committed:
Merge remote-tracking branch 'origin/dev' into dev

Commit:
Add soap fault pre check.

Commit:
Merge branch 'dev' of github.com:spryker-eco/arvato-rss into dev

profuel committed:
ECO-807 tune for PHP files

Commit:
Allow error message be empty.

Commit:
Merge branch 'dev' of github.com:spryker-eco/arvato-rss into dev

Commit:
Get message from soap fault with getter

Commit:
Mock call to arvato rss in functional test.

Commit:
Fix issues reported by phpcs.

Commit:
Fix phpcs warnings that left manually.

Commit:
Fix phpmd warnings.

Commit:
ECO-967. The first version of transfer declaration.

Commit:
ECO-967. Add storeOrder method to api adapter.

Commit:
ECO-967. Add converters for StoreOrder call and factory to born converters.

Commit:
ECO-967. Add return types to interface methods.

Commit:
ECO-967. Update license header.

Commit:
ECO-967. Update return type.

Commit:
ECO-967. Store order exception class.

Commit:
ECO-967. Fix method name.

Commit:
EC967. Fix return type.

Commit:
ECo-967. Extend store order response transfer with error data.

Commit:
ECO-967. Extend api adapter with store order call.

Commit:
ECO-967. Basic response converter for store order.

Commit:
ECO-967. Add mappers, facade method and factory methods.

Commit:
ECO-967. Add post check plugin for store order.

Commit:
ECO-967. Fix arguments.

Commit:
ECO-967. Add needed getters to module config.

Commit:
ECO-967. Add aspect mappers.

Commit:
ECO-967. Remove comment.

Commit:
ECO-967. Remove redundant use.

Commit:
ECO-967. Mapping refactoring.

Commit:
Update license badge.

Commit:
ECO-967. Remove todo comment.

Commit:
ECO-966. Adjust arvato rss related transfers structure inside quote.

Commit:
ECO-967. Fix phpcs and phpmd warnings.

Commit:
Update travis files.

Commit:
ECO-967. Update compoer autoload sections to psr-4.

Commit:
ECO-967. Add databuilder.xml.

Commit:
ECO-967. Add _support with several mocks.

Commit:
ECO-967. Add codeception.yml files.

Commit:
ECO-967. Remove legacy support folder.

Commit:
ECO-967. Remove legacy tests.

Commit:
ECO-967. Test changes.

 - Fixed converter classes.
 - Facade, mapper and converter tests refactored using databuilders.
 - Mapper tests have been created for storeOrder.

Commit:
ECO-967. Fix plugin namespace.

Commit:
ECO-967. Fix test namespaces and names.

Commit:
ECO-967. Replace Store::getInstance call with facade injection.

Commit:
ECO-967. Improve converters.

Commit:
ECO-967. Improve tests.

Commit:
ECO-967. Fix wsdl path for tests.

Commit:
ECO-967. Update tests.

Commit:
ECO-967. Add payment type constant file.

Commit:
ECO-967. Add payment type related constants.

Commit:
ECO-967. Add missed fields to store order request.

Commit:
ECO-967. Code style fixes.

Commit:
ECO-967. Fix const docblock.

Commit:
ECO-967. Update const docblock

Commit:
ECO-967. Insert empty line before return.

Commit:
ECO-967. Fix return type.

Commit:
ECO-967. Remove variable name from docblock.

Commit:
ECO-967. Move store order specific setters from common mapper.

Commit:
ECO-967. Add spryker/sales dependency to communication layer.

Commit:
ECO-967. Remove post check plugin.

Commit:
ECO-967. Use order transfer for store order request. Update mappers.

Commit:
ECO-967. Change argument type.

Commit:
ECO-967. Rename constants file.

Commit:
ECO-967. Add success result code constant.

Commit:
ECO-967. Extend ArvatoRssStoreOrderResponse transfer.

Commit:
ECO-967. Extend store order response converter.

Commit:
ECO-967. Update constans file name in test.

Commit:
ECO-967. Fix bridge.

Commit:
ECO-967. Add transaction logger.

Commit:
ECO-967. Add oms condition and command for store order + transaction logging.

Commit:
ECO-967. Fix code style issues.

Commit:
ECO-967. Fix phpmd warnings.

Commit:
ECO-967. Fix transaction type for risk check logging.

Commit:
ECO-967. Fix tests.

Commit:
ECO-967. Remove redundant use statement.

Commit:
ECO-967. SoapApiAdapter refactoring.

- Separate call objects have been created for each api call.
- Persistance layer related things have been renamed.

Commit:
ECO-967. Rename query, used to get log data.

Commit:
ECO-967. Modify call constants.

Commit:
ECO-967. Remove store order response mapper.

Commit:
ECO-967. Refactoring of store facade order test.

Commit:
ECO-965. Fix code style.

Commit:
ECO-967. Fix travis config filename.

Commit:
ECO-967. Update config.dist.php.

Commit:
ECO-967. Map registerOrder parameter with customer's isGuest property.

Commit:
ECO-967. Fix missed use statement.

Commit:
ECO-967. Update test after refactoring registeredOrder mapping.

Commit:
ECO-967. Rename constants file to *Config.

Commit:
ECO-967. Fix wrong namespaces.

Commit:
ECO-967. Fix phpcs errors.

Commit:
ECO-967. Fix namespace.

Commit:
ECO-967. Fix wrong variable name in test.

Commit:
ECO-967. Add transfer objects for Order and CustomerBilling mappers.

Commit:
ECO-967. Refactoring mappers.

Commit:
ECO-967. Fix register order mapping.

Commit:
ECO-967. Fix mapper tests.

Commit:
ECO-967. Add communication token field to order transfer.

ievgenvarava committed:
ECO-1017: Added shipping address to RiskCheck request, fixed tests.

ievgenvarava committed:
ECO-1017: Added testcase for DeliveryCustomerMapper.

ievgenvarava committed:
ECO-1017: Fixed code sniffer warnings

ievgenvarava committed:
ECO-1017: fixed code review suggestions.

ievgenvarava committed:
Merge pull request #7 from spryker-eco/feature/ECO-1017-shipping-address-in-risk-check

Feature/eco 1017 shipping address in risk check

ievgenvarava committed:
Merge pull request #6 from spryker-eco/feature/store-order

ECO-967: Feature/store order

ievgenvarava committed:
ECO-1016: Provided support for address validation.

ievgenvarava committed:
Merge pull request #8 from spryker-eco/feature/ECO-1016-address-validation

ECO-1016: Provided support for address validation.
